tests/system/aiplatform/test_e2e_forecasting.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from google.cloud import aiplatform
+from google.cloud.aiplatform.compat.types import job_state
+from google.cloud.aiplatform.compat.types import pipeline_state
+import pytest
+from tests.system.aiplatform import e2e_base
+
+_TRAINING_DATASET_BQ_PATH = (
+    "bq://bigquery-public-data:iowa_liquor_sales_forecasting.2020_sales_train"
+)
+_PREDICTION_DATASET_BQ_PATH = (
+    "bq://bigquery-public-data:iowa_liquor_sales_forecasting.2021_sales_predict"
+)
+
+
+@pytest.mark.usefixtures("prepare_staging_bucket", "delete_staging_bucket")
+class TestEndToEndForecasting(e2e_base.TestEndToEnd):
+    """End to end system test of the Vertex SDK with forecasting data."""
+
+    _temp_prefix = "temp-vertex-sdk-e2e-forecasting"
+
+    def test_end_to_end_forecasting(self, shared_state):
+        """Builds a dataset, trains models, and gets batch predictions."""
+
+        # Collection of resources generated by this test, to be deleted during
+        # teardown.
+        shared_state["resources"] = []
```

Comment:
Sorry, we're deprecating the use of `shared_state`. Can you put your test in a `try` block and do the cleanup in `finally`?